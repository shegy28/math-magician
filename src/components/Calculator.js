import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({ total: 0, next: null, operation: null });
  const { total, next, operation } = state;

  const btnClicked = (e) => {
    const clickedBtn = e.target.innerText;
    const gotValue = calculate(state, clickedBtn);
    setState(gotValue);
  };

  return (
    <div className="calculator">
      <div className="result">
        <span>
          {total}
          {' '}
          {operation}
          {' '}
          {next}
        </span>
      </div>
      <div className="num">
        <button type="button" onClick={btnClicked}>AC</button>
        <button type="button" onClick={btnClicked}>+/-</button>
        <button type="button" onClick={btnClicked}>%</button>
        <button type="button" className="act" onClick={btnClicked}>รท</button>
      </div>
      <div className="num">
        <button type="button" onClick={btnClicked}>7</button>
        <button type="button" onClick={btnClicked}>8</button>
        <button type="button" onClick={btnClicked}>9</button>
        <button type="button" className="act" onClick={btnClicked}>x</button>
      </div>
      <div className="num">
        <button type="button" onClick={btnClicked}>6</button>
        <button type="button" onClick={btnClicked}>5</button>
        <button type="button" onClick={btnClicked}>4</button>
        <button type="button" className="act" onClick={btnClicked}>-</button>
      </div>
      <div className="num">
        <button type="button" onClick={btnClicked}>3</button>
        <button type="button" onClick={btnClicked}>2</button>
        <button type="button" onClick={btnClicked}>1</button>
        <button type="button" className="act" onClick={btnClicked}>+</button>
      </div>
      <div className="zero">
        <button type="button" className="o" onClick={btnClicked}>0</button>
        <button type="button" className="zb" onClick={btnClicked}>.</button>
        <button type="button" className="zb act" onClick={btnClicked}>=</button>
      </div>

    </div>
  );
};

export default Calculator;
