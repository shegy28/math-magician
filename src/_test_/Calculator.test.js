import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Calculator from '../components/Calculator';
import '@testing-library/jest-dom';

test('expect home component to match snapshot', () => {
  const view = renderer.create(<Calculator />).toJSON();
  expect(view).toMatchSnapshot();
});

test('Checks a calculate role', () => {
  render(<Calculator />);
  const linkElement = screen.getByRole('Calculate');
  expect(linkElement).toBeInTheDocument();
});

test('renders Ac in the UI', () => {
  render(<Calculator />);
  const linkElement = screen.getByText(/AC/i);
  expect(linkElement).toBeInTheDocument();
});
