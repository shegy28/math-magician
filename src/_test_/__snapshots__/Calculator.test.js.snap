// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect home component to match snapshot 1`] = `
<div
  className="calc-container"
  role="Calculate"
>
  <div
    className="calc-text"
  >
    <h2>
      Let's Do Some Maths
    </h2>
  </div>
  <div
    className="calculator"
  >
    <div
      className="result"
    >
      <span>
        0
         
         
      </span>
    </div>
    <div
      className="num"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="act"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="num"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="act"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="num"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="act"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="num"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="act"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="zero"
    >
      <button
        className="o"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="zb"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="zb act"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
